themes/digital.gov/src/js/gh-edit-commit.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Sets the most recent commit timestamp for the page
+ * This is displayed in #page-data section at the bottom of the page
+ */
+
+// eslint-disable-next-line func-names
+(function () {
+  const editPageLink = document.querySelector("#page-data .gh-commit-date");
+  const gitRepo = {
+    filepath: "",
+    branch: null,
+  };
+
+  /**
+   * Get hugo file path from the data-gh-edit-page attribute to link to Github repo location
+   * example string: news/2023/07/2023-07-19-gsa-shared-service-provider-program-guide.md
+   */
+  function getFilepath() {
+    if (editPageLink) {
+      gitRepo.filepath = document
+        .querySelector("div[data-gh-edit-page]")
+        .getAttribute("data-gh-edit-page");
+    }
+  }
+
+  /**
+   * Set the branch from the URL path
+   * If on cloud.pages get the branch name from the URL
+   * Otherwise, use main for localhost and production
+   */
+  function getBranch() {
+    const host = window.location.hostname;
+    if (host.includes("sites.pages.cloud.gov")) {
+      // eslint-disable-next-line prefer-destructuring
+      gitRepo.branch = window.location.pathname.split("/")[4];
+    } else {
+      gitRepo.branch = "main";
+    }
+  }
+
+  /**
+   * format github ISO date format to human friendly datetime string
```

Comment:
Can we use full sentences for descriptions? Makes it easier to read.

Reply:
Comment is updated to read better.